Add code quality report to CoverityChecker (#147)
melexis/warnings-plugin: README.rst

.. code-block:: bash

    # command line log file

>     mlx-warnings --coverity cov-run-desktop-output.txt

Author: Resolved in 47926867fa1828ec5aefcb9e35ea6b1b665d3ef5

    python -m mlx.warnings --coverity --command <commandforcoverity>

We utilize `cov-run-desktop` in the following manner, where the output is saved in `coverity.log`:

.. code-block:: bash

    cov-run-desktop --text-output-style=oneline --exit1-if-defects false --triage-attribute-regex "classification" ".*" <coverity_files> | tee coverity.log

Subsequently, we process the `coverity.log` file with the mlx-warnings plugin.
The plugin uses a configuration file (`warnings_coverity.yml`) and produces two outputs:
a text file (`warnings_coverity.txt`) and a code quality JSON file (`coverity_code_quality.json`).

.. code-block:: bash

    mlx-warnings --config warnings_coverity.yml -o warnings_coverity.txt -C coverity_code_quality.json coverity.log

This is an example of the configuration file:

> .. code-block:: yml

Reviewer: ```suggestion
.. code-block:: yaml
```
https://pygments.org/docs/lexers/#pygments.lexers.data.YamlLexer

  `column title <Exporting Polyspace Results_>`_ in lowercase as the variable name.
  The default template is ``Polyspace: $check``.

Coverity

>   Named groups of the regular expression can also be used, next to environment variables.
>   The capturing groups are: `path`, `line`, `column`, `curr`, `max`, `checker` and `classification`.
>   `curr` and `max` represents respectively the current warning and the maximum amount of warnings with the same CID.
>   These are always the same if you take the note of `Parse for Coverity Defects`_ into account.

Reviewer: ```suggestion
  Named groups of the regular expression can be used as variables.
  Useful names are: `checker` and `classification`.
```
The path, line and column info are already stored as separate fields for the CQ report. I don't see a use case for curr and max, so I suggest leaving it out of the documentation. What do you think?

Author: Yes, that's clearer
